README
NAME
    Net::Blogger - an OOP-ish interface for accessing a weblog via the
    Blogger XML-RPC API.

SYNOPSIS
     use Net::Blogger;
     my $b = Net::Blogger->new(appkey=>APPKEY);

     $b->BlogId(BLOGID);
     $b->Username(USERNAME);
     $b->Password(PASSWORD);

     $b->BlogId($b->GetBlogId(blogname=>'superfoobar'));

     # Get recent posts

     my ($ok,@p) = $b->getRecentPosts(numposts=>20);

     if (! $ok) {
       croak $b->LastError();
     }

     map { print "\t $_->{'postid'}\n"; } @p;

     # Post from a file

     my ($ok,@p) = $b->PostFromFile(file=>"/usr/blogger-test");

     if (! $ok) {
       croak $b->LastError();
     }

     # Deleting posts

     map {
       $b->deletePost(postid=>"$_") || croak $b->LastError();
     } @p;

     # Getting and setting templates

     my $t = $b->getTemplate(type => 'main');
     $b->setTemplate(type=>'main',template=>\$t) || croak $b->LastError();

     # New post

     my $txt = "hello world.";
     my $id = $b->newPost(postbody=>\$txt) || croak $b->LastError();

     # Get post

     my $post = $b->getPost($id) || croak $b->LastError();
     print "Text for last post was $post->{'content'}\n";

DESCRIPTION
    Blogger.pm provides an OOP-ish interface for accessing a weblog via the
    Blogger XML-RPC API.

ENGINES
    Blogger.pm relies on "engines" to implement it's functionality. The
    Blogger.pm package itself is little more than a wrapper file that
    happens to use a default "Blogger" engine is none other is specified.

       my $manila = Net::Blogger->new(engine=>"manila");

    But wait!, you say. It's an API that servers implements and all I should
    have to do is changed the login data. Why do I need an engine?

    Indeed. Every server pretty much gets the spirit of the API right, but
    each implements the details slightly differently. For example :

    The MovableType XML-RPC server follows the spec for the *getRecentPost*
    but because of the way Perl auto-vivifies hashes it turns out you can
    slurp all the posts for a blog rather than the just the 20 most recent.

    The Userland Manila server doesn't support the *getUsersBlogs* method;
    the Userland RadioUserland server does.

    The Blogger server imposes a limit on the maximum length of a post.
    Other servers don't. (Granted the server in question will return a
    fault, if necessary, but Blogger.pm tries to do the right thing and
    check for these sorts of things before adding to the traffic on the
    network.)

    Lots of weblog-like applications don't support the Blogger API but do
    have a traditional REST interface. With the introduction of Blogger.pm
    "engines", support for these applications via the API can be added with
    all the magic happening behind the curtain, so to speak.

PACKAGE METHODS
  __PACKAGE__->new(\%args)
    Instantiate a new Blogger object.

    Valid arguments are :

    *   engine (required)

        String. Default is "blogger".

    *   appkey

        String. The magic appkey for connecting to the Blogger XMLRPC
        server.

    *   blogid

        String. The unique ID that Blogger uses for your weblog

    *   username

        String. A valid username for blogid

    *   password

        String. A valid password for the username/blogid pair.

    Releases prior to Net::Blogger 0.85 accepted a list of arguments rather
    than a reference. Version 0.85+ are backwards compatible.

    Returns an object. Woot!

  __PACKAGE__->init()
    Initializes the specified engine

Blogger API METHODS
  $pkg->getUsersBlogs()
    Fetch the *blogid*, *url* and *blogName* for each of the Blogger blogs
    the current user is registered to.

    Returns an array ref of hashes.

  $pkg->newPost(\%args)
    Add a new post to the Blogger server.

    Valid arguments are :

    *   postbody

        Scalar ref. *required*

    *   publish

        Boolean.

    If the length of *postbody* exceeds maximum length allowed by the
    Blogger servers -- 65,536 characters -- currently the text will be
    chunked into smaller pieces are each piece will be posted separately.

    Returns an array containing one, or more, post ids.

  $pkg->getPost($postid)
    Returns a hash ref, containing the following keys : userid, postid,
    content and dateCreated.

  $pkg->getRecentPosts(\%args)
    Fetch the latest (n) number of posts for a given blog. The most recent
    posts are returned first.

    Valid arguments are

    *   numposts

        Int. If no argument is passed to the method, default is 1.

        "NumberOfPosts is limited to 20 at this time. Let me know if this
        gets annoying. Letting this number get too high could result in some
        expensive db access, so I want to be careful with it." --Ev

    Releases prior to Net::Blogger 0.85 accepted a list of arguments rather
    than a reference. Version 0.85+ are backwards compatible.

    Returns true or false, followed by an array of hash refs. Each hash ref
    contains the following keys : postid,content,userid,dateCreated

  $pkg->editPost(\%args)
    Update the Blogger database. Set the body of entry $postid to $body.

    Valid arguments are :

    *   postbody (required)

        Scalar ref or a valid filehandle.

    *   postid (required)

        String.

    *   publish

        Boolean.

    If the length of *postbody* exceeds maximum length allowed by the
    Blogger servers -- 65,536 characters -- currently the text will be
    chunked into smaller pieces are each piece will be posted separately.

    Releases prior to Net::Blogger 0.85 accepted a list of arguments rather
    than a reference. Version 0.85+ are backwards compatible.

    Returns an array containing one, or more, post ids.

  $pkg->deletePost(\%args)
    Delete a post from the Blogger server.

    Valid arguments are

    *   postid (required)

        String.

    *   publish

        Boolean.

    Releases prior to Net::Blogger 0.85 accepted a list of arguments rather
    than a reference. Version 0.85+ are backwards compatible.

    Returns true or false.

  $pkg->setTemplate(\%args)
    Set the body of the template matching type *$type*.

     "template is the HTML (XML, whatever -- Blogger can output any sort of text).
      Must contain opening and closing <Blogger> tags to be valid and accepted."
         --Evan

    Valid arguments are

    *   template

        Scalar ref. *required*

    *   type

        String. *required*

        Valid types are "main" and "archiveIndex"

    Releases prior to Net::Blogger 0.85 accepted a list of arguments rather
    than a reference. Version 0.85+ are backwards compatible.

    Returns true or false.

  $pkg->getTemplate(\%args)
    Fetch the body of the template matching type *$type*.

    Valid types are

    *   type

        String. *required*

        Valid types are "main" and "archiveIndex"

    Releases prior to Net::Blogger 0.85 accepted a list of arguments rather
    than a reference. Version 0.85+ are backwards compatible.

    Returns a string.

EXTENDED METHODS
  $pkg->GetBlogId(\%args)
    Return the unique blogid for *$args-*{'blogname'}>.

    Valid arguments are

    *   blogname

        String.

    Releases prior to Net::Blogger 0.85 accepted a list of arguments rather
    than a reference. Version 0.85+ are backwards compatible.

    Returns a string. If no blogname is specified, the current blogid for
    the object is returned.

  $pkg->DeleteAllPosts(\%args)
    Delete all the posts on a weblog. Valid arguments are :

    *   publish

        Boolean.

    Releases prior to Net::Blogger 0.85 accepted a list of arguments rather
    than a reference. Version 0.85+ are backwards compatible.

  $pkg->PostFromFile(\%args)
    Open a filehandle, and while true, post to Blogger. If the length of the
    amount read from the file exceeds the per-post limit assigned by the
    Blogger servers -- currently 65,536 characters -- the contents of the
    file will be posted in multiple "chunks".

    Valid arguments are

    *   file

        /path/to/file *required*

    *   postid

        String.

    *   publish

        Boolean.

    *   tail

        Boolean.

        If true, the method will not attempt to post data whose length
        exceeds the limit set by the Blogger server in the order that the
        data is read. Translation : last in becomes last post becomes the
        first thing you see on your weblog.

    If a *postid* argument is present, the method will call the Blogger API
    *editPost* method with postid. Otherwise the method will call the
    Blogger API *newPost* method.

    Releases prior to Net::Blogger 0.85 accepted a list of arguments rather
    than a reference. Version 0.85+ are backwards compatible.

    Returns true or false, followed by an array of zero, or more, postids.

  $pkg->PostFromOutline(\%args)
    Like *PostFromFile*, only this time the file is an outliner document.

    This method uses Simon Kittle's Text::Outline::asRenderedHTML method for
    posting. As of this writing, the Text::Outline package has not been
    uploaded to the CPAN. See below for a link to the homepage/source.

    Valid outline formats are OPML, tabbed text outline, Emacs' outline-mode
    format, and the GNOME Think format.

    Valid arguments are

    *   file

        /path/to/file *required*

    *   postid

        String.

    *   publish

        Boolean.

    If a *postid* argument is present, the method will call the Blogger API
    *editPost* method with postid. Otherwise the method will call the
    Blogger API *newPost* method.

    Releases prior to Net::Blogger 0.85 accepted a list of arguments rather
    than a reference. Version 0.85+ are backwards compatible.

    Returns true or false, followed by an array of zero, or more, postids.

NOTES
  The Atom API
    In January 2004, Blogger announced their support for the Atom API.

    As of this writing (version 0.87) this package does not support the Atom
    API. If you need to do things Atom-ish, your best bet is to use the
    XML::Atom package.

  Content negotiation
    Persons trying to connect to a server using shortened URLs and content
    negotiation should not be surprised if they encounter weirdness and/or
    errors. Specifically, a HTTP 406 error.

    Some preliminary investigation suggests that, if there's a bug at play
    here, it's a bug somewhere deep in SOAP::Lite/HTTP::* land.

    Patches are welcome. Otherwise, you've been warned. :-)

    See also :

AUTHORS
        Originally authored by Aaron Straup Cope
        Adopted by Christopher H. Laco

SEE ALSO
    Net::Blogger::API::Core

    Net::Blogger::Engine::Base

    http://plant.blogger.com/api/

BUGS
    Hopefully, few. Please reports all bugs to :

     http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Net::Blogger

SOURCE
    You can get the latest version of the source code from the Subversion
    repository at http://handelframework.com/svn/CPAN/Net-Blogger/

LICENSE
    Copyright (c) 2001-2005 Aaron Straup Cope.

    This is free software, you may use it and distribute it under the same
    terms as Perl itself.



Changes
Revision history for Perl extension Net::Blogger.

1.01 Fri Dec 2 18:10:35 2005
    - Added pod coverages, spelling, and syntex tests
    - Added MANIFEST.SKIP
    - Added META.yml to MANIFEST
    - Added metaWeblog.getRecentPosts to Userland::metaWeblog API

1.0	 - Fri Dec 02, 2005

	 - The Nothing Special release - it's just been 4 years
	   and Blogger is finally doing the 2.0/Atom thing for
	   real :

	    http://code.blogspot.com/archives/atom-docs.html

	   I think we can call it 1.0, don't you?

	 - Respect $ENV{http_proxy} (RT ticket 8580)

	 - Updated POD

	 - Replace ExtUtils::MakeMaker with Module::Build

         - Pass maintainership on to CLACO

0.87	 - Tue Feb 10, 2004

	 - Added NOTES concerning content-negotiation (RT bug
	   #2615) and the Atom API

	 - Updated POD

0.86	 - Tue Jul 15 2003

	 - Blessed as 0.86

	 - Wondered whether to bless it as 1.0 now that Pyra/Google
	   has said they're no longer actively developing the API.
	   Time enough for that, I guess...

0.86_02	 - Mon Jul 14 2003

	 - Added support for the getRecentPostTitles() and
           publishPost() methods in ::Engine::Movabletype::mt.pm

	 - Added fix to regex(p) in ::Engine::Base.pm used to
	   determine parent API engine to make 'warnings' shut
	   up. (thanks andi!)

	 - Added more error reporting to t/00-basic.t

	 - Updated POD

0.86_01	 - Wed May 22 2003

	 - Updated POD in ::Userland::metaWeblog to indicated
	   required 'postid' for editPost()

	 - Fixed typo in ::Engine::Base.pm

	 - Added [y/n] to ask_yesno() prompt in 00-basic.t

	 - (props to Morbus Iff for pointing most of these things
	   out to me)

0.85	 - Wed Feb 05 2003

	 - Fixed struct/hash bugs in ::metaWeblog.pm when passing
	   arguments to SOAP::Lite (props to mystery rt.cpan user
	   for the bug report)

	 - Updated methods to accept hash references where appropriate

	 - Removed CHANGES from individual POD files because it was
	   just a pain in the ass to keep track of that kind of thing
	   in more than one place.

	 - Replaced test.pl with t/00-basic.t

	 - Updated POD (and sure everything returns OK w/ podchecker)

0.8.4	 - Fri Jan 10 2003

	 - Added support for the metaWeblog.newMediaObject method

	 - Updated POD

0.8.3	 - Mon Sep 23 2002

	 - Updated docs and POD (props to a nice person name 'rho'
	   who was good enough to carefully read the docs and submit
	   bug reports)

0.8.2	 - Mon Sep 02 2002

	 - Added support for the metaWeblog API to the Manila engine

	 - Modified Net::Blogger::Engine::Blogger to allow
	   user-defined Proxy URI

	 - Updated Net::Blogger::Engine::Base to undef _client only
	   when a new Proxy is defined

	 - Updated Net::Blogger::Engine::Userland to undef _client
	   when a new Proxy is defined

	 - Moved $pkg->metaWeblog from Net::Blogger::Engine::Radio.pm
	   into ::Engine::Userland.pm

	 - Updated POD

0.8.1	 - Thu Jun 17 2002

	 - Added support for the mt.getTrackbackPings method in
	   Net::Blogger::Engine::Movabletype::mt.pm

0.8	 - Fri May 31 2002

	 - Added Net::Blogger::Engine::Slash and
	   Net::Blogger::Engine::Slash::slashcode

	 - Added check_* methods to Net::Blogger::API::Core

	 - Added calls to check_* methods, to Engine::* files where
	   necessary

	 - Removed left-over deletePost method from Blogger.pm (gah!)

	 - Added modules required by slashcode.pm to Makefile.PL

	 - Added &Net::Blogger::Engine::Userland::BlogId to ensure
	   that blogid contains a trailing slash.

	 - Updated POD

0.7	 - Sat May 04 2002

	 - Added hooks to Net::Blogger::Engine::Base::_Type to deal with
	   arrays and hashes correctly

	 - Updated POD in Net::Blogger::Engine::Movabletype::mt

0.7_b1	 - Sat May 04 2002

	 - Added support for the metaWeblog API in the MovableType
	   engine

	 - Added support for the mt (MovableType) API in the MovableType
	   engine

	 - Added hooks in Net::Blogger::Base to filter changes to login
	   data in a parent engine to child APIs

	 - Added hooks in Net::Blogger::Engine::Userland::metaWeblog to
	   catch calls to methods not supported by the Movabletype engine.
	   Ugh.

	 - Added quotes to $VERSION in all packages.

	 - Updated POD in Movabletype.pm

	 - Upped version requirement for SOAP::Lite to 0.55

0.6.3	 - Mon Apr 15 2002

	 - Went through all the packages and fixed POD errors (ugh.)

0.6.2.2	 - Sun Apr 14 2002

	 - No changes. But, since I uploaded the current codebase
	   to the CPAN as v 0.6.1.2, I decided to upload a new
	   version with a new version number just to keep everyone
           on first.

0.6.2.1


   Updated POD

0.6.2


   Added support for the UserLand metaWeblog API in
   the RadioUserLand engine

0.6.1


   Bugs fixes in Blogger::Engine::Base

0.6


   Moved most of the code in to Blogger::API::Core,
   Blogger::API::Extended, Blogger::Engine::Base and
   Blogger::Engine::Blogger

   Replaced use of Frontier::Client with XMLRPC::Lite
   (in Blogger::Engine::Base)

   Updated POD

0.5.1


   Modified internals to load implementation specific
   subclass based on the engine argument passed to
   the constructor. Props to Simon Kittles for the
   smack upside the head about the right way to do
   this :-)

   Updated POD.

0.5


   Added Blogger API getPost method.

   Updated POD.

0.4.6.1


   Added Blogger.pm BLOGGER_PROXY constant.

   Added Blogger.pm Proxy and MaxPostPostLength
   accessors for corresponding constants. Previously,
   these values were either read from a scalar
   constant or an AUTOLOAD method. The change allows
   [insert blogger-mimicking interface here]
   subclasses to override the methods and specify an
   approriate value.

   Updated POD

0.4.5


    Added Blogger API getRecentPosts method.

    Updated POD

0.4.4.1


   Clarified a few error messages;

   Fixed remaining instances of
   "Error::Simple->record() and return0. in
   &AUTOLOAD



0.4.4


    Added use of Error.pm

    Added Blogger.pm LastError method.


    Wrapped Frontier::Client method calls in eval

lib/Net/Blogger/API/Extended.pm
{

=head1 NAME

Net::Blogger::API::Extended - provides helper methods not defined in the Blogger API.

=head1 SYNOPSIS

 It's very dark in here because this is a black box.

=head1 DESCRIPTION

This package is inherited by I<Net::Blogger::Engine::Base> and provides helper
methods not defined in the Blogger API.

=cut

package Net::Blogger::API::Extended;
use strict;

$Net::Blogger::API::Extended::VERSION   = '1.0';
@Net::Blogger::API::Extended::ISA       = qw ( Exporter );
@Net::Blogger::API::Extended::EXPORT    = qw ();
@Net::Blogger::API::Extended::EXPORT_OK = qw ();

use Exporter;
use FileHandle;

=head1 OBJECT METHODS

=head2 $pkg->MaxPostLength

Abstract method for returning the max post length

=cut

sub MaxPostLength {
    return undef;
}

=head2 $pkg->GetBlogId(\%args)

Return the unique blogid for I<$args{'blogname'}>.

Valid arguments are

=over

=item *

B<blogname> => string.

=back

Releases prior to Net::Blogger 0.85 accepted a list of arguments
rather than a reference. Version 0.85+ are backwards compatible.

Returns a string. If no blogname is specified, the current blogid for
the object is returned.

=cut

sub GetBlogId {
    my $self = shift;
    my $args = (ref($_[0]) eq "HASH") ? shift : { @_ };

    my $blogid = undef;

    if (! $args->{'blogname'}) {
	return $self->{'_blogid'};
    }

    my $blogs = $self->getUsersBlogs()
	|| return undef;

    foreach my $b (@$blogs) {
	if ($b->{'blogName'} eq $args->{'blogname'}) {
	    $blogid = $b->{'blogid'};
	    last;
	}
    }

    return $blogid;
}

=head2 $pkg->DeleteAllPosts(\%args)

Delete all the posts on a weblog. Valid arguments are :

=over 4

=item *

B<publish>

Boolean.

=back

Releases prior to Net::Blogger 0.85 accepted a list of arguments
rather than a reference. Version 0.85+ are backwards compatible.

Returns true or false.

=cut

sub DeleteAllPosts {
    my $self = shift;
    my $args = { @_ };

    my ($ok,@pids) = $self->getRecentPosts(numposts=>20);

    while (@pids) {
	foreach my $p (@pids) {
	    $self->deletePost(postid=>$p->{'postid'},publish=>$args->{'publish'});
	}

	($ok,@pids) = $self->getRecentPosts(numposts=>20);
    }

    return $ok;
}

=head2 $pkg->PostFromFile(\%args)

Open a filehandle, and while true, post to Blogger. If the length of the amount
read from the file exceeds the per-post limit assigned by the Blogger servers --
currently 65,536 characters -- the contents of the file will be posted in multiple
"chunks".

Valid arguments are

=over

=item *

B<file>

/path/to/file I<required>

=item *

B<postid>

String.

=item *

B<publish>

Boolean.

=item *

B<tail>

Boolean.

If true, the method will not attempt to post data whose length exceeds the limit
set by the Blogger server in the order that the data is read. Translation : last
in becomes last post becomes the first thing you see on your weblog.

=back

If a I<postid> argument is present, the method will call the Blogger API I<editPost>
method with postid. Otherwise the method will call the Blogger API I<newPost> method.

Releases prior to Net::Blogger 0.85 accepted a list of arguments
rather than a reference. Version 0.85+ are backwards compatible.

Returns true or false, followed by an array of zero, or more, postids.

=cut

sub PostFromFile {
    my $self = shift;
    my $args = (ref($_[0]) eq "HASH") ? shift : { @_ };

    if (! -f $args->{'file'}) {
      $self->LastError("Not a file.");
	return 0;
    }

    my $fh = FileHandle->new();
    my $ok = $fh->open("<$args->{'file'}");

    if (! $ok) {
      $self->LastError("Failed to open file : $!");
	return 0;
    }

    my $method  = ($args->{'postid'}) ? "editPost" : "newPost";

    if (! $args->{'tail'}) {

	local $/;
	undef $/;

	my $postbody = <$fh>;
	$fh->close();

	return $self->$method(postbody=>\$postbody,%$args);
    }

    my $post    = "";
    my @postids = ();

    while (<$fh>) {
	my $line = $_;
	chomp $line;

	$post  .= $line;
	my $len = length($post);

	if (($self->MaxPostLength()) && ($len > $self->MaxPostLegth())) {

	    my $postbody  = substr($post,0,$self->MaxPostLength());
	    my $remainder = $self->_TrimPostBody(\$postbody);

	    my ($pid) = $self->$method(
				       postbody => \$postbody,
				       postid   => $args->{'postid'},
				       publish  => $args->{'publish'},
				       );

	    if (! $pid) {
		$fh->close();

	      $self->LastError("Encountered an error posting. Exiting prematurely.");
		return (0,@postids);
	    }

	    push(@postids,$pid);
	    $post = $remainder.substr($post,$self->MaxPostLength(),$len);
	}
    }

    $fh->close();

    if (! $post) {
      $self->LastError("Failed to read any data from file.");
	return 0;
    }

    my ($pid) = $self->$method(
			       postbody => \$post,
			       postid   => $args->{'postid'},
			       publish  => $args->{'publish'},
			       );

    if (! $pid) {
      $self->LastError("Encountered an error posting last chunk : ".Error->prior());
	return (0, @postids);
    }

    push (@postids,$pid);
    return (1,@postids);
}

=head2 $pkg->PostFromOutline(\%args)

Like I<PostFromFile>, only this time the file is an outliner document.

This method uses Simon Kittle's Text::Outline::asRenderedHTML method for
posting. As of this writing, the Text::Outline package has not been uploaded
to the CPAN. See below for a link to the homepage/source.

Valid outline formats are OPML, tabbed text outline, Emacs' outline-mode format,
and the GNOME Think format.

Valid arguments are

=over

=item *

B<file>

/path/to/file I<required>

=item *

B<postid>

String.

=item *

B<publish>

Boolean.

=back

If a I<postid> argument is present, the method will call the Blogger API I<editPost>
method with postid. Otherwise the method will call the Blogger API I<newPost> method.

Releases prior to Net::Blogger 0.85 accepted a list of arguments
rather than a reference. Version 0.85+ are backwards compatible.

Returns true or false, followed by an array of zero, or more, postids.

=cut

sub PostFromOutline {
    my $self = shift;
    my $args = (ref($_[0]) eq "HASH") ? shift : { @_ };

    my $class = "Text::Outline";

    # We'll get rid of this when, and if, a proper
    # Makefile is ever written...
    eval "require $class"
	|| &{ $self->LastError($@); return 0; };

    my $outline = $class->new(load=>$args->{'file'})
	|| &{ $self->LastError($!); return 0; };

    my $postbody = $outline->asRenderedHTML();

    my $method   = ($args->{'postid'}) ? "editPost" : "newPost";

    return $self->$method(postbody=>\$postbody,%$args);
}

sub _PostInChunks {
    my $self = shift;
    my $args = (ref($_[0]) eq "HASH") ? shift : { @_ };

    my $caller  = (caller(1))[3];
    my @chunks  = ();
    my @postids = ();

    unless ($caller =~ /^(Net::Blogger::Base::API::)(editPost|newPost)$/) {
	$self->LastError("$caller is not a valid caller for this method.");
	return 0;
    }

    my $text = $args->{"postbody"};

    while ( $$text ) {
	my $chunk     = substr($$text,0,$self->MaxPostLength());
	my $remainder = $self->_TrimPostBody(\$chunk);

	$$text = $remainder.substr($$text,length($chunk),length($$text));

	# Since Blogger posts are chronological, we add chunks
	# to the top of the stack. That way, the 'end' pieces get
	# added first and, in the end, the text will be displayed
	# in the order it was written.
	# 20010813 (asc)

	unshift (@chunks, \$chunk);
    }

    map {
	$args->{"postbody"} = $_;
	push(@postids, $self->$caller(%$args));
    } @chunks;

    return @postids;
}

sub _TrimPostBody {
    my $self = shift;
    my $body = shift;

    if (ref($body) ne "SCALAR") {
	$self->LastError("Input must be a scalar ref.");
	return undef;
    }
}

=head1 VERSION

1.0

=head1 DATE

$Date: 2005/03/26 19:29:08 $

=head1 AUTHOR

Aaron Straup Cope

=head1 SEE ALSO

L<Net::Blogger::Engine::Base>

L<Net::Blogger::API::Core>

=head1 LICENSE

Copyright (c) 2001-2005 Aaron Straup Cope.

This is free software, you may use it and distribute it under the same terms as Perl itself.

=cut

return 1;

}
